intellij-plugin-verifier/verifier-test/src/test/java/com/jetbrains/pluginverifier/verifiers/resolution/MethodsTest.kt

```diff
@@ -37,6 +39,145 @@ class MethodsTest {
+      return
+    }
+    if (linkedList !is ResolutionResult.Found) {
+      fail("Class '$linkedListBinaryName' not found in the mock plugin by resolver")
+      return
+    }
+    val methodName = "set"
+    val arrayListSetAsmMethod = arrayList.findMethodByName(methodName)
+    val linkedListSetAsmMethod = linkedList.findMethodByName(methodName)
+
+
+    if (arrayListSetAsmMethod == null) {
+      fail("Method '$methodName' not found in '$arrayListBinaryName'")
+      return
+    }
+
+    if (linkedListSetAsmMethod == null) {
+      fail("Method '$methodName' not found in '$linkedListBinaryName'")
+      return
+    }
+    val arrayListMethod = MethodAsm(arrayList.asClassFile(), arrayListSetAsmMethod)
+    val linkedListSetMethod = MethodAsm(linkedList.asClassFile(), linkedListSetAsmMethod)
+
+    sameParameters(arrayListMethod, linkedListSetMethod)
+  }
+
+  @Test
+  fun `method parameter count does not match`() {
+    val arrayListBinaryName = "java/util/ArrayList"
+
+    val resolver = createJdkResolver()
+    // E set(int,E)
+    val set = resolver.findMethodInClass(arrayListBinaryName, "(ILjava/lang/Object;)Ljava/lang/Object;")
+    // int size()
+    val size = resolver.findMethodInClass(arrayListBinaryName, "()I")
+
+    assertFalse(sameParameters(set, size))
+  }
+
+  @Test
+  fun `method parameters do not match`() {
+    val arrayListBinaryName = "java/util/ArrayList"
+
+    val resolver = createJdkResolver()
+    // E set(int,E)
+    val set = resolver.findMethodInClass(arrayListBinaryName, "(ILjava/lang/Object;)Ljava/lang/Object;")
+    // public List<E> subList(int fromIndex, int toIndex)
+    val subList = resolver.findMethodInClass(arrayListBinaryName, "(II)Ljava/util/List;")
+
+    assertFalse(sameParameters(set, subList))
+  }
+
+  @Test
+  fun `method parameters override with the same parameters`() {
+    val abstractCollectionBinaryName = "java/util/AbstractCollection"
+    val objectBinaryName = "java/lang/Object"
+
+    val toStringDesc = "()Ljava/lang/String;"
+
+    val resolver = createJdkResolver()
+    val arrayListToString = resolver.findMethodInClass(abstractCollectionBinaryName, toStringDesc)
+    val objectToString = resolver.findMethodInClass(objectBinaryName, toStringDesc)
+
+    assertTrue(arrayListToString.isOverriding(objectToString, resolver))
+  }
+
+  @Test
+  fun `method parameters override with the covariancy`() {
+    val pkg = "mock/plugin/overrideOnly/covariant"
+
+    val childName: BinaryClassName = "$pkg/Child"
+    val parentName: BinaryClassName = "$pkg/Parent"
+
+    val getIndexerInChild = "()L$pkg/ChildResult;"
+    val getIndexerInParent = "()L$pkg/ParentResult;"
+
+    val resolver = createTestResolver()
+    val arrayListToString = resolver.findMethodInClass(childName, getIndexerInChild)
+    val objectToString = resolver.findMethodInClass(parentName, getIndexerInParent)
+
+    assertTrue(arrayListToString.isOverriding(objectToString, resolver))
+  }
+
+
+  /*
+  @Test
```

Comment:
???

Reply:
Oh, this was a very unfortunate leftover. Rewrote the unit test to have proper and matching variable names.